src/control/cmd/daos_agent/security_rpc.go

```diff
@@ -8,20 +8,40 @@ package main
 
 import (
 	"context"
+	"fmt"
 	"net"
 	"os/user"
+	"time"
 
 	"github.com/pkg/errors"
 
 	"github.com/daos-stack/daos/src/control/drpc"
+	"github.com/daos-stack/daos/src/control/lib/cache"
 	"github.com/daos-stack/daos/src/control/lib/daos"
 	"github.com/daos-stack/daos/src/control/logging"
 	"github.com/daos-stack/daos/src/control/security"
 	"github.com/daos-stack/daos/src/control/security/auth"
 )
 
 type (
-	credSignerFn func(*auth.CredentialRequest) (*auth.Credential, error)
+	// credSignerFn defines the function signature for signing credentials.
+	credSignerFn func(context.Context, *auth.CredentialRequest) (*auth.Credential, error)
+
+	// credentialCache implements a cache for signed credentials.
+	credentialCache struct {
+		log          logging.Logger
+		cache        *cache.ItemCache
+		credLifetime time.Duration
+		cacheMissFn  credSignerFn
+	}
+
+	// cachedCredential wraps a cached credential and implements the cache.ExpirableItem interface.
+	cachedCredential struct {
+		cacheItem
+		key       string
+		expiredAt time.Time
```

Comment:
```suggestion
		expiresAt time.Time
```
Just a nit but `expiredAt` reads to me as "already expired"